compiler/rustc_expand/src/mbe/macro_parser.rs

```diff
@@ -324,8 +324,7 @@ pub(super) fn count_names(ms: &[TokenTree]) -> usize {
                 TokenTree::Delimited(_, ref delim) => count_names(&delim.tts),
                 TokenTree::MetaVar(..) => 0,
                 TokenTree::MetaVarDecl(..) => 1,
-                // FIXME(c410-f3r) MetaVarExpr should be handled instead of being ignored
-                // https://github.com/rust-lang/rust/issues/9390
+                // Meta-variable expressions can not appear in the LHS
                 TokenTree::MetaVarExpr(..) => 0,
```

Comment:
Panicking will abort execution because of
https://github.com/rust-lang/rust/blob/f8a29bd95e0427ea1a46412ee2bb48e92644edc9/compiler/rustc_expand/src/mbe/quoted.rs#L215-L216

In other words, RHS meta-variable expressions eventually end-up here.

I kept the `0` return because this specific snippet only counts meta-variables.

Reply:
Please, leave a comment explaining why it's `0` and not an assert.